Makefile

```diff
@@ -165,7 +165,7 @@ go-mod-tidy/%: DIR=$*
 go-mod-tidy/%:
 	@echo "$(GO) mod tidy in $(DIR)" \
 		&& cd $(DIR) \
-		&& $(GO) mod tidy -compat=1.21
+		&& $(GO) mod tidy -compat=$(GO_VERSION)
```

Comment:
We can simply remove `-compat=`. It is a NOOP if the version in `go.mod` is the same.

Reply:
I have removed -compat and GO_VERSION (no longer used) with https://github.com/open-telemetry/opentelemetry-proto-go/pull/213/commits/18e4e749d358c9876853274c3c30567a52e3df6b.